src/components/PrimaryPanes/SourcesTree.js

```diff
@@ -353,15 +362,21 @@ class SourcesTree extends Component<Props, State> {
   }
 }
 
-export default connect(
-  state => {
-    return {
-      shownSource: getShownSource(state),
-      selectedSource: getSelectedSource(state),
-      debuggeeUrl: getDebuggeeUrl(state),
-      expanded: getExpandedState(state),
-      projectRoot: getProjectDirectoryRoot(state)
-    };
-  },
-  dispatch => bindActionCreators(actions, dispatch)
-)(SourcesTree);
+const mapStateToProps = state => {
+  return {
+    shownSource: getShownSource(state),
+    selectedSource: getSelectedSource(state),
+    debuggeeUrl: getDebuggeeUrl(state),
+    expanded: getExpandedState(state),
+    projectRoot: getProjectDirectoryRoot(state)
+  };
+};
+
+const mapDispatchToProps = dispatch => {
+  return {
+    setExpandedState: () => dispatch(setExpandedState()),
```

Comment:
Redux automatically binds dispatch to an object of functions so we could make this look like
```js
const actionCreators = {
  setExpandedState,
  selectSource
};

export default connect(mapStateToProps, actionCreators)(SourcesTree);
```

Reply:
This is also a great way to do it!